webapp/webapp_flow.go

```diff
@@ -64,7 +63,7 @@ func (flow *Flow) BrowserURL(baseURL string, params BrowserParams) (string, erro
 	ru.Host = fmt.Sprintf("%s:%d", ru.Hostname(), flow.server.Port())
 	flow.server.CallbackPath = ru.Path
 	flow.clientID = params.ClientID
-	flow.redirectURI = params.RedirectURI
+	flow.redirectURI = ru.String()
```

Comment:
What difference happens with this?  It seems strange it would be needed

Reply:
We use the `redirectURI` in two places:
First to get code: https://github.com/pluralsh/oauth/blob/main/webapp/webapp_flow.go#L71 where the real address is set (ru.String()  contains a random port on which the server is listening)
Second to get a token: https://github.com/pluralsh/oauth/blob/main/webapp/webapp_flow.go#L111 where a predefined address is used (without the correct port number) from here: 
```
	flow := &oauth.Flow{
		CallbackURI:  "http://127.0.0.1/callback",
		ResponseType: "code",
	}
```

So the two calls have different callback addresses. This PR set the same redirect URI for both calls

Reply:
I have tested this change also with Gitlab because we use the same flow, and works fine

Reply:
ok sounds good